rcl/src/rcl/node.c

```diff
@@ -179,7 +211,47 @@ rcl_node_init(
   }
   // actual domain id
   node->impl->actual_domain_id = domain_id;
-  node->impl->rmw_node_handle = rmw_create_node(name, local_namespace_, domain_id);
+
+  const char * ros_enable_security = NULL;
+  const char * ros_enforce_security = NULL;
+
+  if (rcutils_get_env(sros_enable_security_var_name, &ros_enable_security)) {
```

Comment:
Given that getenv is used in various places in the codebase already. I think that implementing a pseudo thread-safe version of it goes along refactoring all calls to get env throughout the codebase. So I'm going to submit this and the associated PRs for review right now and iterate on the getenv improvement/refactor in another set of PRs.